app/models/token.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class Token
+  attr_reader :token_string
+
+  def initialize(token_string)
+    @token_string = token_string
+    validate_token
+  end
+
+  def to_s
+    @token_string
+  end
+
+  def payload
+    @payload ||= if @token_string
+                   pubkey = expected_key
+                   return if pubkey.blank?
+
+                   JWT.decode(@token_string, pubkey, true, algorithm: 'RS256')[0]
+                 end
+  rescue JWT::ExpiredSignature
+    raise error_klass('Validation error: token has expired')
+  end
+
+  def expected_key
```

Comment:
Unless I'm misunderstanding what this does, I believe this method extracts the `kid` from our JWT, then looks up + returns the associated public key from Okta. What about just `public_key`?